scipy/optimize/_linprog.py

```diff
@@ -135,742 +141,1009 @@ def linprog_terse_callback(xk, **kwargs):
+            # test_unbounded_no_nontrivial_constraints_1
+            # test_unbounded_no_nontrivial_constraints_2
+            status = 3
+            message = ("The problem is (trivially) unbounded "
+                       "because there are no non-trivial constraints and "
+                       "a) at least one decision variable is unbounded "
+                       "above and its corresponding cost is negative, or "
+                       "b) at least one decision variable is unbounded below "
+                       "and its corresponding cost is positive. ")
+        else:  # test_empty_constraint_2
+            status = 0
+            message = ("The solution was determined in presolve as there are "
+                       "no non-trivial constraints.")
+        complete = True
+        x[c < 0] = ub_mod[c < 0]
+        x[c > 0] = lb_mod[c > 0]
+        # where c is zero, set x to a finite bound or zero
+        x_zero_c = ub_mod[c == 0]
+        x_zero_c[np.isinf(x_zero_c)] = ub_mod[c == 0][np.isinf(x_zero_c)]
+        x_zero_c[np.isinf(x_zero_c)] = 0
+        x[c == 0] = x_zero_c
+        # if this is not the last step of presolve, should convert bounds back
+        # to array and return here
+
+    # *sigh* - convert bounds back to their standard form (list of tuples)
+    # again, in retrospect, numpy array would be standard form
+    lb[np.equal(lb, -np.inf)] = None
+    ub[np.equal(ub, np.inf)] = None
+    bounds = np.hstack((lb[:, np.newaxis], ub[:, np.newaxis]))
+    bounds = bounds.tolist()
+    for i, row in enumerate(bounds):
+        for j, col in enumerate(row):
+            if str(
+                    col) == "nan":  # comparing col to float("nan") and
+                                    # np.nan doesn't work. should use np.isnan
+                bounds[i][j] = None
+
+    # remove redundant (linearly dependent) rows from equality constraints
+    n_rows_A = A_eq.shape[0]
+    redundancy_warning = ("A_eq does not appear to be of full row rank. To "
+                          "improve performance, check the problem formulation "
+                          "for redundant equality constraints.")
+    if (sps.issparse(A_eq)):
+        if rr and A_eq.size > 0:  # TODO: Fast sparse rank check?
+            A_eq, b_eq, status, message = _remove_redundancy_sparse(A_eq, b_eq)
+            if A_eq.shape[0] < n_rows_A:
+                warn(redundancy_warning, OptimizeWarning)
+            if status != 0:
+                complete = True
+        return (c, c0, A_ub, b_ub, A_eq, b_eq, bounds,
+                x, undo, complete, status, message)
+
+    # This is a wild guess for which redundancy removal algorithm will be
+    # faster. More testing would be good.
+    small_nullspace = 5
+    if rr and A_eq.size > 0:
+        try:  # TODO: instead use results of first SVD in _remove_redundancy
+            rank = np.linalg.matrix_rank(A_eq)
+        except Exception:  # oh well, we'll have to go with _remove_redundancy_dense
```

Comment:
Why would matrix_rank throw exception?

Reply:
I think one of the SVD routines bugs out?

Reply:
I can take care of that if you have any examples that raises exceptions.

Reply:
Good to know. If we start working on presolve again I'll keep it in mind. But I don't have any examples right now unfortunately. Not sure why I didn't report them at the time, but they probably only occurred when I was really abusing it. 

Any interest on looking into something with SuperLU? #7700 is much more important, I think.

Reply:
Ouch. It is indeed. I'll check.

Reply:
@ilayn Thank you! If that would work, I think my sparse LU rank-one update code would work. That would enable a sparse version of the pivot-based redundancy removal routine and a revised simplex linear programming solver. And then, eventually, I could finally finish the MILP solver - which was the whole reason I got into this mess from the start : )